packages/sp/files/types.ts

```diff
@@ -103,8 +103,8 @@ export class _Files extends _SharePointQueryableCollection<IFileInfo[]> {
     @tag("fis.addChunked")
     public async addChunked(url: string, content: Blob, progress?: (data: IFileUploadProgressData) => void, shouldOverWrite = true, chunkSize = 10485760): Promise<IFileAddResult> {
 
-        const info: { ServerRelativeUrl: string } = await spPost(this.clone(Files, `add(overwrite=${shouldOverWrite},url='${escapeQueryStrValue(url)}')`, false));
-        const file = File(`_api/web/getFileByServerRelativeUrl('${info.ServerRelativeUrl}')`);
+        const response: IFileInfo = await spPost(this.clone(Files, `add(overwrite=${shouldOverWrite},url='${escapeQueryStrValue(url)}')`, false));
+        const file = File(odataUrlFrom(response));
```

Comment:
The following code uploads large file to a site collection called "anothersitecollection", and the code is executed from the root site collection: the upload will fail because there is a bug when creating the File object, it always default to the current context Web/Site, not the one what was created. With this fix it created the file object correctly.
```typescript
const web = Web("https://contoso.sharepoint.com/sites/anothersite");
const uploadedfile = await web.getFolderByServerRelativeUrl("/sites/anothersite/Shared%20Documents")
         .files.addChunked( '50MB.zip', file, data => { console.log("uploading"); }, true, 5000000);
```
Same function **approodataUrlFrom(response)** is used in method [finishUpload](https://github.com/pnp/pnpjs/blob/version-2/packages/sp/files/types.ts#L523), which is part of the .addCunked method.

Reply:
AH, I see now thanks - brain wasn't braining that. Cool, this looks good then, will get it merged for the next release. Thank you for the fix!